corehq/apps/geospatial/models.py

```diff
@@ -1 +1,130 @@
-# Create your models here.
+from math import radians, cos, sin, asin, sqrt
+from shapely.geometry import Point
+import geopandas as gpd
+
+
+class GeoObject:
+    """
+    `lat` and `lon` are assumed to be in a EPSG:4326 map projection.
+    """
+    def __init__(self, id, lon, lat):
+        if lon < -180 or lon > 180:
+            raise ValueError("Invalid lon value. Must be [-180, 180]")
```

Comment:
Can we maybe have a base exception class for this app, just to keep the exceptions scoped? Something like `GeoSpatialException`. We can then create subclasses from this for exceptions like this. In this case, maybe something like `InvalidCoordinate(GeoSpatialException)`.

Not sure how others feel about this?

Reply:
I simply used the `ValueError` exception as it represents when incorrect arguments are given, which felt appropriate in this scenario. Your idea sounds good however, as it could help us to take more useful/appropriate action later down the line if we are able to catch more specific exceptions.